app/src/main/java/com/kylecorry/trail_sense/astronomy/infrastructure/commands/SunsetAlarmCommand.kt

```diff
@@ -45,88 +45,22 @@ class SunsetAlarmCommand(private val context: Context) : CoroutineCommand {
         val tomorrowSunset =
             astronomyService.getTomorrowSunTimes(location.location, suntimesMode).set
 
-        if (todaySunset != null) {
-            when {
-                isPastSunset(todaySunset) -> {
-                    // Missed the sunset, schedule the alarm for tomorrow
-                    setAlarm(
-                        tomorrowSunset?.minus(alertDuration)
-                            ?: todaySunset.plusDays(1)
-                    )
-                }
+        todaySunset ?: return@onDefault
 
-                withinAlertWindow(todaySunset, alertDuration) -> {
-                    // Send alert, schedule alarm for tomorrow's sunset or else at some point tomorrow
-                    sendNotification(todaySunset)
-                    setAlarm(
-                        tomorrowSunset?.minus(alertDuration)
-                            ?: todaySunset.plusDays(1)
-                    )
-                }
-
-                else -> { // Before the alert window
-                    // Schedule alarm for sunset
-                    setAlarm(todaySunset.minus(alertDuration))
-                }
-            }
-        } else {
-            // There isn't a sunset today, schedule it for tomorrow
-            setAlarm(tomorrowSunset?.minus(alertDuration) ?: now.plusDays(1))
-        }
-    }
-
-    private fun isPastSunset(sunset: ZonedDateTime): Boolean {
-        return ZonedDateTime.now().isAfter(sunset)
-    }
-
-    private fun withinAlertWindow(sunset: ZonedDateTime, alertDuration: Duration): Boolean {
-        val alertTime = sunset.minus(alertDuration)
-        val minAlertTime = alertTime.minus(alertWindow)
-        val alertRange = Range(minAlertTime, sunset)
-        return alertRange.contains(ZonedDateTime.now())
-    }
-
-    private fun sendNotification(sunset: ZonedDateTime) {
-
-        val lastSentDate = userPrefs.astronomy.sunsetAlertLastSent
-        if (LocalDate.now() == lastSentDate) {
-            return
+        if (isAfterSunset(todaySunset)) {
+            // Missed the sunset, schedule the alarm for tomorrow
+            setAlarm(
+                tomorrowSunset?.minus(alertDuration)
+                    ?: todaySunset.plusDays(1)
+            )
+            return@onDefault
```

Comment:
Looks like there were some issues with this file - I'm not sure I prefer the direction you were heading though, I recommend restoring what was originally in place (the when block)